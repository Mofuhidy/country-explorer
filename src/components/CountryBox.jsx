import { useNavigate } from "react-router-dom";
import PropTypes from "prop-types";

import { FiArrowRightCircle } from "@react-icons/all-files/fi/FiArrowRightCircle";
function CountryBox({ flag, name, capital, pop, id, alt }) {
  const navigate = useNavigate();

  return (
    <>
      <li
        className="country-box col-span-1 text-white p-4 relative flex flex-col"
        id={id}>
        <button type="button" onClick={() => navigate(`/conutrydata/${id}`)}>
          <FiArrowRightCircle className=" absolute right-2 top-1 text-xl" />
        </button>
        <div className=" h-20 w-20 self-center">
          <img src={flag} alt={alt} />
        </div>
        <div className="smallDetails self-end">
          <p className="countryName font-bold uppercase text-right text-xl">
            {name}
          </p>
          <p className="countryName font-light text-sm text-right">{capital}</p>
          <p className="population  text-sm">
            population: <span className=" font-bold text-base">{pop}</span>
          </p>
        </div>
      </li>
    </>
  );
}

CountryBox.propTypes = {
  flag: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  capital: PropTypes.arrayOf(PropTypes.string),
  pop: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
};

export default CountryBox;
