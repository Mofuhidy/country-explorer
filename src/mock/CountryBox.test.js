import { render, screen } from "@testing-library/react";
import React from 'react';
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import CountryBox from "../components/CountryBox";
import store from "../redux/store";
import "@testing-library/jest-dom";

it("Check if the component changed", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <CountryBox />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Check if the component CountryBox container is there", async () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <CountryBox />
      </BrowserRouter>
    </Provider>
  );
  const container = await screen.findByTestId("box");
  expect(container).toBeInTheDocument();
});
