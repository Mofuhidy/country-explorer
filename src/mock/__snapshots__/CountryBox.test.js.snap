// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the component changed 1`] = `
<li
  className="country-box col-span-1 text-white p-4 relative flex flex-col"
  data-testid="box"
>
  <button
    onClick={[Function]}
    type="button"
  >
    <svg
      className=" absolute right-2 top-1 text-xl"
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <polyline
        points="12 16 16 12 12 8"
      />
      <line
        x1="8"
        x2="16"
        y1="12"
        y2="12"
      />
    </svg>
  </button>
  <div
    className=" h-20 w-20 self-center"
  >
    <img />
  </div>
  <div
    className="smallDetails self-end"
  >
    <p
      className="countryName font-bold uppercase text-right text-xl"
    />
    <p
      className="countryName font-light text-sm text-right"
    />
    <p
      className="population  text-sm"
    >
      population:
       
      <span
        className=" font-bold text-base"
      />
    </p>
  </div>
</li>
`;
